src/extension/executors/runner.ts

```diff
@@ -119,15 +119,31 @@ export async function executeRunner(
     program.handleInput(input)
   })
 
-  const useIntegrated = isRunmeIntegratedTerminalEnabled()
+  const revealNotebookTerminal = interactive ?
+    background ?
+      isNotebookTerminalFeatureEnabled('backgroundTask') :
+      isNotebookTerminalFeatureEnabled('interactive') :
+    isNotebookTerminalFeatureEnabled('nonInteractive')
 
   if (!interactive) {
     const output: Buffer[] = []
 
     const mime = mimeType || 'text/plain' as const
 
+    if (revealNotebookTerminal) {
```

Comment:
This is effectively duplicated code, see the previous review. You can bring this block outside of the if block.